portalnet/src/overlay_service.rs

```diff
@@ -758,58 +758,146 @@ where
+                            );
+                        });
+                    }
+                    FindContentQueryResult::Utp {
+                        connection_id,
+                        peer,
+                        nodes_to_poke,
+                    } => {
+                        let metrics = self.metrics.clone();
+                        let utp = self.utp_socket.clone();
+                        let id = u16::from_be(connection_id);
+                        let source = match self.find_enr(&peer) {
+                            Some(enr) => enr,
+                            _ => {
+                                warn!("Received uTP payload from unknown peer");
+                                return;
+                            }
+                        };
+                        let cid = utp_rs::cid::ConnectionId {
+                            recv: id,
+                            send: id.wrapping_add(1),
+                            peer: crate::discovery::UtpEnr(source),
+                        };
+                        let validator = self.validator.clone();
+                        let store = self.store.clone();
+                        let kbuckets = self.kbuckets.clone();
+                        let command_tx = self.command_tx.clone();
+                        tokio::spawn(async move {
+                            metrics.report_utp_active_inc(UtpDirectionLabel::Inbound);
+                            let mut stream =
+                                match utp.connect_with_cid(cid.clone(), UTP_CONN_CFG).await {
+                                    Ok(stream) => stream,
+                                    Err(err) => {
+                                        metrics.report_utp_outcome(
+                                            UtpDirectionLabel::Inbound,
+                                            UtpOutcomeLabel::FailedConnection,
+                                        );
+                                        warn!(
+                                            %err,
+                                            cid.send,
+                                            cid.recv,
+                                            peer = ?cid.peer.client(),
+                                            "Unable to establish uTP conn based on Accept",
+                                        );
+                                        return;
+                                    }
+                                };
+
+                            let mut data = vec![];
+                            if let Err(err) = stream.read_to_eof(&mut data).await {
+                                metrics.report_utp_outcome(
+                                    UtpDirectionLabel::Inbound,
+                                    UtpOutcomeLabel::FailedDataTx,
+                                );
+                                error!(%err, cid.send, cid.recv, peer = ?cid.peer.client(), "error reading data from uTP stream");
```

Comment:
We almost never want to have identical log lines in multiple places in the code. Otherwise, tracking down a surprising log is harder, because it has multiple possible causes.

Plus, it seems like an opportunity to add more context, like "stream, while handling a FindContent query".